### PR TITLE
Start refactor of peep code to label known addresses

### PR DESCRIPTION
**Pull request by duncanspumpkin**
2 bugs fixed. Incorrect sprite image for chicken and juice

### CODE REVIEW

**Comment by Nubbie** [comment]
Fixes #4545?

File: src/openrct2/peep/peep.c [hunk 1]
```diff
@@ -3772,9 +3684,11 @@ static void peep_update_ride_sub_state_15(rct_peep* peep){
 	x *= 32;
 	y *= 32;
 
-	const rct_xy16 edx  = _97e1bc[ride->type][peep->var_37];
-	x += edx.x;
-	y += edx.y;
+	assert(ride->type == RIDE_TYPE_SPIRAL_SLIDE);
+	const rct_xy16 slidePlatformDestination  = SprialSlideWalkingPath[peep->var_37];
```
Comment (marijnvdwerf) [inline src/openrct2/peep/peep.c]: `Sprial`

File: src/openrct2/peep/peep_data.c [hunk 2]
```diff
@@ -1841,7 +1841,7 @@ static const uint8 PeepSpriteImage_9855C4_35_unk04[] = { 13, 0, 1, 2, 2, 2, 2, 3
 static const uint8 PeepSpriteImage_9855C4_36_unk04[] = { 30, 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 10, 10, 10, 11, 12, 11, 12, 11, 12, 11, 12, 11, 11, 11, 11, 11, 13, 14, 15, 0xFF };
 
 
-static const rct_sprite_image PeepSpriteImage_982888[] = {
+static const rct_sprite_image PeepSpriteImage_Normal[] = {
 	{ 6409, PeepSpriteImage_982888_0_unk04 },
```
Comment (marijnvdwerf) [inline src/openrct2/peep/peep_data.c]: Guessing these numbers are all sprite ID's? :/

File: src/openrct2/peep/peep.c [hunk 3]
```diff
@@ -1730,37 +1638,37 @@ typedef struct item_pref {
...
+		{ 0, PEEP_ITEM_HOT_DOG, PEEP_SPRITE_TYPE_HOT_DOG  },
+		{ 0, PEEP_ITEM_TENTACLE, PEEP_SPRITE_TYPE_TENTACLE },
+		{ 0, PEEP_ITEM_CANDY_APPLE, PEEP_SPRITE_TYPE_TOFFEE_APPLE },
+		{ 0, PEEP_ITEM_DONUT, PEEP_SPRITE_TYPE_DONUT },
+		{ 1, PEEP_ITEM_PRETZEL, PEEP_SPRITE_TYPE_PRETZEL },
+		{ 1, PEEP_ITEM_COOKIE, PEEP_SPRITE_TYPE_PRETZEL },
+		{ 1, PEEP_ITEM_CHOCOLATE, PEEP_SPRITE_TYPE_COFFEE },
+		{ 1, PEEP_ITEM_ICED_TEA, PEEP_SPRITE_TYPE_COFFEE },
+		{ 1, PEEP_ITEM_FUNNEL_CAKE, PEEP_SPRITE_TYPE_FUNNEL_CAKE },
+		{ 1, PEEP_ITEM_BEEF_NOODLES, PEEP_SPRITE_TYPE_NOODELS },
```
Comment (HaasJona) [inline src/openrct2/peep/peep.c]: Noodels ➡️ Noodels

File: src/openrct2/peep/peep_data.c (hunk 2 above)
Comment (duncanspumpkin) [inline src/openrct2/peep/peep_data.c]: Yeah I worked out what it all represents. This is all related to the peep actions first action is just none so its all the walking and such. 6409 is the base sprite id for sitting. 0_unk04 is the the amount you increment the base to get to the next frames sprite id.

File: src/openrct2/peep/peep.c (hunk 3 above)
Comment (duncanspumpkin) [inline src/openrct2/peep/peep.c]: Damn sometimes I wish there was a spell checker to hide the embarrassment.

Comment (HaasJona) [inline src/openrct2/peep/peep.c]: I use Clion as IDE which has a spell checker integrated :smile:

Comment (IntelOrca) [inline src/openrct2/peep/peep.c]: @duncanspumpkin There are various VS extensions, but they can hurt performance sometimes or just be annoying with the amount of technical terms which aren't officially words.

File: src/openrct2/paint/sprite/peep.c [hunk 4]
```diff
@@ -76,8 +76,8 @@ void peep_paint(rct_peep * peep, int imageDirection)
 		spriteType = peep->next_action_sprite_type;
 		imageOffset = 0;
 	}
-
-	uint32 baseImageId = (imageDirection >> 3) + sprite.sprite_image[spriteType].base_image + imageOffset * 4;
+
```
Comment (IntelOrca) [inline src/openrct2/paint/sprite/peep.c]: Trailing whitespace

Comment (duncanspumpkin) [inline src/openrct2/paint/sprite/peep.c]: It's a Hodge podge of formatting right now the whole PR as I had limited time. I wouldn't bother looking at it just yet

File: src/openrct2/peep/peep.h [hunk 5]
```diff
@@ -645,6 +645,9 @@ typedef struct rct_peep_animation_entry {
 	const rct_sprite_bounds *sprite_bounds;    // 0x04
 } rct_peep_animation_entry;
 
+// rct2: 0x00982708
+extern rct_peep_animation_entry g_peep_animation_entries[48];
```
Comment (marijnvdwerf) [inline src/openrct2/peep/peep.h]: Is there some `{{enum}}_COUNT` you can use instead of 48?

Comment (duncanspumpkin) [inline src/openrct2/peep/peep.h]: Yes there is now, peep action sprite.

**Comment by janisozaur** [comment]
Lots of new trailing whitespace, can you clean it up?

**Comment by duncanspumpkin** [comment]
@janisozaur i only found one trailing whitespace is there anything I'm missing?

**Comment by janisozaur** [comment]
@duncanspumpkin fixed it myself. Feel free to meld it to one of your commits.

**Comment by janisozaur** [comment]
@duncanspumpkin I have checked and it looks like you have not only renamed values, but also made changes to them. If that was intentional, this needs `network-version` tag and appropriate change in `network.h`; if not, that change needs to be found.

**Comment by duncanspumpkin** [comment]
The changes made were intentional but it should only affect the display of peeps. I guess it wouldn't harm to do an increment as well.

**Pull request merged**